src/common/errors_js.ml

```diff
@@ -106,6 +201,21 @@ let file_of_error err =
   let loc = loc_of_error err in
   match loc.Loc.source with Some filename -> filename | None -> ""
 
+(* TODO: Is this actually needed? Does it make sense with the new format? *)
+let remove_newlines (color, text) =
+  (color, Str.global_replace (Str.regexp "\n") "\\n" text)
+
+let print_error_color ~one_line ~color error =
+  let level, messages, trace_reasons = error in
+  let messages = append_trace_reasons messages trace_reasons in
+  let messages = merge_comments_into_blames messages in
+  let header = print_error_header (List.hd messages) in
+  let main_file = file_of_error error in
+  let formatted_messages = List.map (print_message_nice main_file) messages in
```

Comment:
Actually I do handle that case :) Look for "See:" line here: https://dl.dropboxusercontent.com/u/4980295/flow/new-flow.html

The idea is if the blame points to a different file we are printing "See: that/other/file.js". This way visually we can pretend it's one "main" file per error, and the error trace can still span different files.